### PR TITLE
feature(EXEC-892): add session schedule_at to the exec command

### PR DESCRIPTION
**Pull request by GregoryEssertel**
Summary:
Use the exec command to test the session in display only mcx and absolute time constraints. Associated lab PR: https://github.com/strateos/lab/pull/1237

If the user uses the exec command, upon success, they will receive a session id. Then can then reuse this session id, to interleave another run on the created worldstate.

The user has also access to the flag --schedule-delay to provide a relative starting time in minutes,
or --schedule-at to provide an absolute starting time ("14:13" meaning today at 2:13pm, "2021-02-20T12:34" meaning at that exact time).

Test Plan:
Add unit testing for the new flags.

### CODE REVIEW

File: transcriptic/commands.py
Comment (yangchoo) [inline transcriptic/commands.py:1528]: nit: Couldn't

File: transcriptic/cli.py
Comment (yangchoo) [inline transcriptic/cli.py:664]: Could you further clarify `Relative time format +NUM in minutes`? Looking at the code, I understand this as if you specify `+15`, that would imply scheduling 15 minutes in the future. 

IMO we should consider breaking it out into a separate option. (e.g. `schedule-after`)
Its a bit confusing since its a different syntax, and we specified absolute time right at the start of the help message before adding this new format at the end.

Comment (yangchoo) [reply transcriptic/cli.py:664]: We should clarify that this time should refer to the scheduling instance's machine time and not the client's machine time (in case of different timezones for example)

File: transcriptic/cli.py
Comment (yangchoo) [inline transcriptic/cli.py:652]: Let's rephrase this slightly to talk about the effect of specifying this. Instead of stating this references a session Id which already exists, let's state what the session id is used for

**Review by yangchoo — COMMENTED** [review]
Please add to CHANGELOG.rst as well.

Have not integration-tested this, but the code itself generally looks fine.

File: transcriptic/commands.py
Comment (yangchoo) [reply transcriptic/commands.py:1528]: Let's also restate the expected format in the error message so that its clearer what went wrong

**Review by yangchoo — CHANGES_REQUESTED** [review]
Integration tested this. `schedule-at` seems to be working as expected. However, seems like `--session-id` args don't seem to be respected yet? This is with the specified lab PR.

E.g.
```
➜ python thermocycle-test.py | transcriptic exec -a "http://localhost:9000" -w wc0 --session-id mcxs13pzvr6vqxxna
Sending request...
Success. View http://localhost:9000/dashboard?sessionId=mcxs13pzvt42ff7vx to see the scheduling outcome.
```

File: transcriptic/cli.py
Comment (GregoryEssertel) [reply transcriptic/cli.py:664]: Using time makes thing complicated!
I will create a different flag for the relative time, seems better. I wanted the option for local testing as it is simpler to do everything in relative if the exact time doesn't matter.

Currently, the absolute time is based on the local time of the client machine. The absolute time is converted to the epoch time for SCLE to deserialize an Instant. Maybe it should be done MCX to make sure that if the user look at the dashboard there is correspondance.

**Comment by GregoryEssertel** [comment]
> Integration tested this. `schedule-at` seems to be working as expected. However, seems like `--session-id` args don't seem to be respected yet? This is with the specified lab PR.
> 
> E.g.
> 
> ```
> ➜ python thermocycle-test.py | transcriptic exec -a "http://localhost:9000" -w wc0 --session-id mcxs13pzvr6vqxxna
> Sending request...
> Success. View http://localhost:9000/dashboard?sessionId=mcxs13pzvt42ff7vx to see the scheduling outcome.
> ```

There were no checks if there were mutually exclusive parameters. In this case the -w took precedence over the --session-id. I fixed that. I will do more integration tests today, I actually started with this PR, and got delayed a little bit on the lab PR.

**Comment by GregoryEssertel** [comment]
Update: when I did the integration tests, I realize that all frontend were getting all the scheduling updates or all thenotifications. So I am adding a session id for those messages to keep the frontend workable!

**Comment by GregoryEssertel** [comment]
The lab PR is currently deployed on the wctest at haven:

```
$ transcriptic preview Read | transcriptic exec --api http://wctest.txhq.net:9000 -w wc4
Sending request...
Success. View http://wctest.txhq.net:9000/dashboard?sessionId=mcxs13q92cytswyxw to see the scheduling outcome.
```

**Comment by GregoryEssertel** [comment]
The lab PR has been merge and will be available in the next deployment.

**Review by scottyantipa — APPROVED** [review]

**Comment by GregoryEssertel** [comment]
@yangchoo it seems that I can't merge without you validating I made the change. Let me know if it is OK!

**Review by yangchoo — APPROVED** [review]
my bad for being a blocker. Will defer to scott's review for testing. Generally lgtm

**Comment by GregoryEssertel** [comment]
No worries, I am not sure how Github handle request for changes? Thanks!

**Pull request merged**
